src/GVF/GVF.jl

```diff
@@ -33,8 +39,15 @@ function get(gvf::GVF, state_t, action_t, state_tp1, action_tp1, preds_tilde)
     return c, γ, π_prob
 end
 
-get(gvf::GVF, state_t, action_t, state_tp1, preds_tilde) =
-    get(gvf::GVF, state_t, action_t, state_tp1, nothing, preds_tilde)
+struct GVFCollection{C<:AbstractCumulant, D<:AbstractDiscount, P<:AbstractPolicy} <: AbstractGVF
+    cumulants::Vector{C}
+    discounts::Vector{D}
+    policies::Vector{P}
+end
```

Comment:
I thought we were going to do a list of GVFs?

Reply:
Is a GVFCollection or GVFHorde an abstractGVF? Seems messy.

Reply:
It uses the same api so I  figured it would be suitable

but also I feel like you can consider a gvf collection to be a gvf in a higher dimensional space, mapping say (s,a,s') to a vector instead of a scalar

Reply:
With the current approach, can we do something like this?
```julia
c = [ CumulantType1(), CumulantType2() ]
d = [ ConstantDiscount(), DecayingDiscount() ]
p = [ PolicyType1(), PolicyType2() ]
gvfc = GVFCollection(c, d, p)
```

My fear is that the type parameters would have to be widened to accept this. That is, the type-param `D` would need to be `AbstractDiscount` instead of a specific type of discount.

Is this a valid concern in Julia, or is this something that the language will comfortably handle?

Reply:
Hmmm. Even if they have some of the same interface I'm not sure it would be good generally.

You make a good point with it being a higher dimensional GVF, but there are several ways of doing a multi-dimensional GVF (not only as a collection of GVFs). Do we want the gvfcollection to be able to have multi-dimensional GVFs in its collection?

Reply:
I'm not sure actually.. We do specify that D<:AbstractDiscount, so you can create vectors with any type as long as it descents from AbstractDiscount, but I'm not sure if this leads to performance issues

Reply:
That is a good question. Could use some testing.